tests/infrastructure/test_kubernetes_job.py

```diff
@@ -892,7 +892,7 @@ def mock_stream(*args, **kwargs):
 
 
 def test_watch_deadline_is_computed_before_log_streams(
-    mock_k8s_client, mock_watch, mock_k8s_batch_client
+    mock_k8s_client, mock_watch, mock_k8s_batch_client, mock_anyio_sleep
```

Comment:
`mock_anyio_sleep` mocks `pendulum.now` not `time.monotonic` fyi — it probably could mock `monotonic` too.